client/source/states/GameState.cpp

```diff
@@ -174,9 +176,20 @@ void GameState::initShaders() {
 
 void GameState::draw(gk::RenderTarget &target, gk::RenderStates states) const {
 	// FIXME: This uniform is not used anymore since water/leaves effects are disabled
-	// gk::Shader::bind(&m_shader);
-	// m_shader.setUniform("u_time", gk::GameClock::getTicks());
-	// gk::Shader::bind(nullptr);
+	// Now used by day/night cycle
+	gk::Shader::bind(&m_shader);
+	m_shader.setUniform("u_time", gk::GameClock::getTicks());
+	gk::Shader::bind(nullptr);
+
+	float pi = 3.1415927;
+	float frequency = 480000;
+	float time = gk::GameClock::getTicks() % 960000;
+	float sunlight = std::min(std::max((double)(1 + std::cos(2 * pi / frequency * time) * 2.0), 0.0), 1.0);
+	float red = std::min(std::max((double)sunlight - 0.55, 0.0), 0.45);
+	float green = std::min(std::max((double)sunlight - 0.35, 0.0), 0.65);
+	float blue = std::min(std::max((double)sunlight - 0.09, 0.0), 0.91);
```

Comment:
why do you need all these `double` casts?

Reply:
they didn't compile because min() and max() saw the decimal numbers as doubles for some reason, but now they're fine when cast is removed